packages/components/src/progressive-disclosure-panel/index.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	createContext,
+	useContext,
+	useEffect,
+	useMemo,
+	useState,
+} from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ProgressiveDisclosurePanelItem from './item';
+import ProgressiveDisclosurePanelTitle from './title';
+
+const PanelContext = createContext( {} );
+
+export const usePanelContext = () => useContext( PanelContext );
+
+const isMenuItem = ( item ) => item.type === ProgressiveDisclosurePanelItem;
+
+const ProgressiveDisclosurePanel = ( props ) => {
+	const { children, className, label: menuLabel, resetAll, title } = props;
+	const [ menuItems, setMenuItems ] = useState( {} );
+
+	// When conditionally including components e.g. { isShown && <Component /> }
```

Comment:
I guess we can update the comment to say the component is only interested in descendant ProgressiveDisclosurePanelItem.

Reply:
Thanks. ✅ Done.